Comb sort

ColeSeverson opened the pull request:
<!-- Enter a brief description of the changes you've made in the next line -->

<!-- Check the following boxes, if applicable, by replacing the space inside
	"[ ]" with an "x", eg. [x] -->
- [X] Read the [contribution guidelines][contrib-guidelines]
- [x] Added [unit tests][unit-tests] (or unit tests have already been added)
- [x] Updated the [Contents][contents]

<!-- If this PR closes an existing issue, write "Closes #123" in the next line,
        where 123 is the issue number (for example) -->
Closes #172. 
Closes #185.


[contrib-guidelines]: https://github.com/faheel/Algos/blob/master/CONTRIBUTING.md
[unit-tests]: https://github.com/faheel/Algos/blob/master/C%2B%2B/UNIT_TESTS.md
[contents]: https://github.com/faheel/Algos/tree/master/C%2B%2B#contents

Commit:
added comb_sort.hpp to the includes

Commit:
Merge branch 'master' of https://github.com/faheel/Algos into comb_sort

faheel commented on C++/include/sorting/comb_sort.hpp:
```diff
@@ -0,0 +1,45 @@
+
+#include "./utils.hpp"
+
+void comb_sort(vector<int>& input, const int order = 1, const bool to_show_state = true){
```
`to_show_state` should be `false` by default.

faheel replied:
This is still pending.

faheel commented on C++/include/sorting/comb_sort.hpp:
```diff
@@ -0,0 +1,45 @@
+	bool sorted = false;
+
+	while(!sorted){
+		gap = (gap / shrink_factor); //This should automatically floor bc it is an int
```
This is simpler:
```c++
gap /= shrink_factor;
```

faheel replied:
This is still pending.

faheel reviewed (changes requested):
@ColeSeverson Thank you for contributing! There are only a few minor things to fix and then it'll be good to merge:
* Re-indent code with 4 spaces per indent level.
* Add a single space between `if`, `for`, `while` etc. and the opening parenthesis `(`. This makes the code easier to read as these statements can be distinguished from function calls more easily.
* Drop the changes in `README.md` that were merged from `master`. It's not required.
* Add an entry for comb sort in the contents (in `C++/README.md`).

Commit:
fixed spacing in Comb_sort.hpp

Commit:
Merge branch 'reorganise' of https://github.com/faheel/Algos into comb_sort

Commit:
added comb_sort.hpp to readme

ColeSeverson commented:
Implemented the fixes that you requested. It should be up to spec now.

ColeSeverson commented:
@faheel Implemented the spacing checks that you wanted added, as well as adding it to the README.md

faheel commented on C++/README.md:
```diff
@@ -38,6 +38,7 @@ Implementation of well-known (and some rare) algorithms, in C++.
   * [Radix sort](source/sorting/radix_sort.cpp)
   * [Selection sort](source/sorting/selection_sort.cpp)
   * [Shell sort](source/sorting/shell_sort.cpp)
+  * [Comb sort](source/include/sorting/comb_sort.hpp)
```
Please maintain alphabetical sorting in the list.

faheel commented on C++/include/sorting/comb_sort.hpp:
```diff
@@ -0,0 +1,45 @@
+/*
+	Comb Sort by Cole Severson
+	---------
+	A sorting algorithm desinged to improve upon bubble sort by by eliminating small values        that start near the end of the list.
```
Looks like a newline got deleted here.

faheel reviewed (changes requested):
A few changes that were requested in the previous review weren't made:
* Add a single space between `if`, `for`, `while` etc. and the opening parenthesis `(`. This makes the code easier to read as these statements can be distinguished from function calls more easily.
* Move `comb_sort.hpp` under `C++/include/algorithm/sorting/`.

Commit:
spacing added between statements and opening parenthesis

Commit:
moved file to include/algorithms/sorting

Commit:
Revert "moved file to include/algorithms/sorting"

This reverts commit f3f0f854f88bb328121812c1d7f1fe0a86e8db8f.

Commit:
Moved comb_sort.hpp into include/algorithms/sorting/

ColeSeverson commented:
@faheel Ok, I moved Comb_sort.hpp to include/algorithms/sorting/ and added spacing around the parenthesis on conditionals and loops. Let me know if anything else needs changing!

Commit:
Fixed alphabetical order in the README.md

ColeSeverson commented:
@faheel Fixed the order in the README.md

ColeSeverson commented:
@faheel All changes should be fixed

alxmjo committed:
Merge branch 'master' into comb_sort

alxmjo committed:
Fix default parameter and simplify division

alxmjo committed:
Replace bc with because

alxmjo commented:
Assuming the tests that are currently running pass, then this should be good to merge.
